### PR TITLE
Samplegen: CLI for Java standalone samples

### PR DESCRIPTION
**Pull request by hzyi-google**
This PR makes Java standalone samples (almost) copy-paste compilable and executable from command line by:
- add imports
- change the existing main method to a static sample method and add a main method that calls the sample method
- handle CLIs using apache cli package
- add client initialization

Remaining work to make for Java standalone samples which will come in a following PR:
- remove the boilerplate code describing each sample
- add license
- correctly import the client (`LibraryClient`)
- put samples to the correct package and directory

### CODE REVIEW

**Comment by hzyi-google** [comment]
PTAL @beccasaurus @pongad @vchudnov-g

File: src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline [hunk 1]
```diff
@@ -7686,30 +7803,68 @@ public class MonologAboutBookCallableCallableStreamingClientProg {
 
 // FIXME: Insert here code to prepare the request fields, make the call, process the response.
 
+package com.google.gcloud.pubsub.v1;
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline]: Why is this in pubsub?

File: src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline [hunk 2]
```diff
@@ -7686,30 +7803,68 @@ public class MonologAboutBookCallableCallableStreamingClientProg {
...
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import com.google.example.library.v1.Book;
+import com.google.example.library.v1.SeriesUuid;
+import com.google.example.library.v1.Shelf;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline]: I think there's some imports missing. We need to import `LibraryClient` right?

File: src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline (hunk 1 above)
Comment (hzyi-google) [reply src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline]: It's defined [here](https://github.com/googleapis/gapic-generator/blob/b4c1fb4c4e24f9cbb32c9c91894c7de17d6ca47b/src/test/java/com/google/api/codegen/testsrc/libraryproto/library_gapic.yaml#L5).
Shall we change the package above to `com.google.cloud.example.library.v1`?
And therefore, the samples will be in the package `com.google.cloud.example.example.library.v1.snippets`, as to be consistent with examples in `google-cloud-java`?

File: src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline (hunk 2 above)
Comment (hzyi-google) [reply src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline]: Yes I'll fix it

File: src/main/resources/com/google/api/codegen/java/standalone_sample.snip [hunk 3]
```diff
@@ -217,6 +241,54 @@
...
+  Options options = new Options();
+  @join param : sample.initCode.argDefaultParams
+    options.addOption(
+        Option.builder("")
+          .required(false)
+          .hasArg(true)
+          .longOpt("{@param.identifier}")
+          .build());
+  @end
+  CommandLine cl = (new DefaultParser()).parse(options, args);
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/java/standalone_sample.snip]: I suggest a blank line before this `CommandLine` line

File: src/main/resources/com/google/api/codegen/java/standalone_sample.snip [hunk 4]
```diff
@@ -24,17 +24,41 @@
...
+        }
         // [END {@sample.regionTag}_core]
       }
-    }
 
+      public static void main(String[] args) {
+        @if sample.initCode.argDefaultParams.size
+          {@processCliArguments(sample)}
+        @end
+        {@sample.sampleFunctionName}({@commaSeparatedArgs(sample.initCode.argDefaultParams)});
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/java/standalone_sample.snip]: I suggest a blank line before this line

**Pull request merged**